cranelift/filetests/src/zkasm_codegen.rs

```diff
@@ -33,13 +33,23 @@ pub fn generate_zkasm(settings: &ZkasmSettings, wasm_module: &[u8]) -> String {
 
     let mut program: Vec<String> = Vec::new();
 
-    let start_func = zkasm_environ
+    let main_func = zkasm_environ
         .info
-        .start_func
-        .expect("Must have a start function");
+        .functions
+        .iter()
+        .find(|&(_, exportable_func)| exportable_func.export_names.contains(&"main".to_string()));
```

Comment:
nit: match a full name. Export order is unspecified and we shouldn't grab something like `maintenance` as the entry point.

nit 2: that `to_string` is superfluous, I believe.

Reply:
The trick is that `export_names` is a `Vec<String>`: https://github.com/near/wasmtime/blob/ba69743b3260fd94e489b771f5637c0dd5ffbf96/cranelift/wasm/src/environ/zkasm.rs#L38
So `contains` is needed.

When I remove `to_string` I get:

```
error[E0308]: mismatched types
    --> cranelift/filetests/src/zkasm_codegen.rs:42:77
     |
42   |         .find(|&(_, exportable_func)| exportable_func.export_names.contains("main"))
     |                                                                    -------- ^^^^^^ expected `&String`, found `&str`
     |                                                                    |
     |                                                                    arguments to this method are incorrect
     |
     = note: expected reference `&std::string::String`
                found reference `&'static str`
```

Not sure how to resolve this without `to_string`.

P.S. I wish we had type annotation intelligence in GitHub :)